app/ts/components/simulationExplaining/customExplainers/GovernanceVoteVisualizer.tsx

```diff
@@ -87,11 +88,87 @@ export function VotePanel({ inputParams }: { inputParams: GovernanceVoteInputPar
 	</>
 }
 
+export type ShowSuccessOrFailureParams = {
+	simTx: SimulatedAndVisualizedTransaction
+	currentBlockNumber: undefined | bigint
+	rpcConnectionStatus: RpcConnectionStatus
+	simulationAndVisualisationResults: SimulationAndVisualisationResults
+	simulateGovernanceContractExecutionReply: SimulateGovernanceContractExecutionReply | undefined
+	renameAddressCallBack: RenameAddressCallBack
+	addressMetaData: readonly AddressBookEntry[]
+}
+
+const simulateGovernanceVote = () => sendPopupMessageToBackgroundPage({ method: 'popup_simulateGovernanceContractExecution' })
+
+const ShowSuccessOrFailure = ({ currentBlockNumber, rpcConnectionStatus, simulateGovernanceContractExecutionReply, simTx, simulationAndVisualisationResults, renameAddressCallBack, addressMetaData }: ShowSuccessOrFailureParams) => { 
+	const missingAbiText = 'The governance contract is missing an ABI. Add an ABI to simulate execution of this proposal.'
+	if (simulateGovernanceContractExecutionReply === undefined) {
+		return <div style = 'display: flex; justify-content: center;'>
+			{ !(simulationAndVisualisationResults.rpcNetwork.httpsRpc === 'https://rpc.dark.florist/birdchalkrenewtip' // todo remove this check
+				|| simulationAndVisualisationResults.rpcNetwork.httpsRpc=== 'https://rpc.dark.florist/winedancemuffinborrow') ? <p class = 'paragraph'> experimental rpc client required </p> : <></> }
+				
+			{ simTx.transaction.to !== undefined && 'abi' in simTx.transaction.to && simTx.transaction.to.abi !== undefined ?
+				<button
+					class = { `button is-primary` }
+					onClick = { simulateGovernanceVote }
+					disabled = { !(simulationAndVisualisationResults.rpcNetwork.httpsRpc === 'https://rpc.dark.florist/birdchalkrenewtip' // todo remove this check
+							|| simulationAndVisualisationResults.rpcNetwork.httpsRpc=== 'https://rpc.dark.florist/winedancemuffinborrow')
+				}>
+					Simulate execution on a passing vote
+				</button>
+			: <> <MissingAbi
+					errorMessage = { missingAbiText }
+					addressBookEntry = { simTx.transaction.to }
+					renameAddressCallBack = { renameAddressCallBack }
+				/>
+			</> }
+		</div>
+	}
+	if (simulateGovernanceContractExecutionReply.data.success == false) {
+		return <div style = 'display: grid; grid-template-rows: max-content' >
+			{ simulateGovernanceContractExecutionReply.data.error.type === 'MissingAbi' ? <MissingAbi
+				errorMessage = { missingAbiText }
+				addressBookEntry = { simulateGovernanceContractExecutionReply.data.error.addressBookEntry }
+				renameAddressCallBack = { renameAddressCallBack }
+			/> : <ErrorComponent warning = { false } text = { simulateGovernanceContractExecutionReply.data.error.message }/> }
```

Comment:
`ErrorComponent` accepts `undefined` for `warning` props which can be omitted

```suggestion
			/> : <ErrorComponent text = { simulateGovernanceContractExecutionReply.data.error.message }/> }
```